src/Parsers/ParserCreateQuery.cpp

```diff
@@ -222,17 +223,69 @@ bool ParserProjectionDeclaration::parseImpl(Pos & pos, ASTPtr & node, Expected &
     return true;
 }
 
+bool ParserForeignKeyDeclaration::parseImpl(Pos & pos, ASTPtr & node, Expected & expected)
```

Comment:
We already have (almost) the same code: https://github.com/ClickHouse/ClickHouse/blob/1d170f57457fd575714797615b65f2304746e06c/src/Parsers/MySQL/ASTDeclareReference.cpp#L46-L102

Let's reuse it and avoid code duplication.